cli/compose/loader/loader.go

```diff
@@ -481,6 +485,45 @@ func transformExternal(data interface{}) (interface{}, error) {
 	}
 }
 
+func transformServicePort(data interface{}) (interface{}, error) {
+	switch entries := data.(type) {
+	case []interface{}:
+		// We process the list instead of individual items here.
+		// The reason is that one entry might be mapped to multiple ServicePortConfig.
+		// Therefore we take an input of a list and return an output of a list.
+		ports := []interface{}{}
+		for _, entry := range entries {
+			switch value := entry.(type) {
+			case int:
+				v, err := toServicePortConfigs(fmt.Sprint(value))
+				if err != nil {
+					return data, err
+				}
+				for _, vv := range v {
+					ports = append(ports, vv)
+				}
+			case string:
+				v, err := toServicePortConfigs(value)
+				if err != nil {
+					return data, err
+				}
+				for _, vv := range v {
+					ports = append(ports, vv)
+				}
```

Comment:
Thanks @dnephin. There was an compilation error when appending directly:
```
cli/compose/loader/loader.go:502: cannot use v (type []"github.com/docker/docker/cli/compose/types".ServicePortConfig) as type []interface {} in append
```

Now the `toServicePortConfigs(value string)` has be changed to return `[]interface{}` (instead of `[]types.ServicePortConfig`) so it is possible to append directly with the update.